icechunk/src/session.rs

```diff
@@ -220,6 +271,7 @@ impl Session {
             snapshot_id,
             change_set: ChangeSet::default(),
             default_commit_metadata: SnapshotProperties::default(),
+            splits: Default::default(),
```

Comment:
Note to self: Interesting, I wonder how this thing will get initialized with the info from previous sessions

Reply:
@dcherian maybe add a comment here explaining how the splits get lazily populated